### ISSUE TITLE
The crate doesn't support `wasm/32-bit targets`

### ISSUE THREAD

**Issue by niklasad1** [paritytech/scale-decode#4]
`$ cargo check --target wasm32-unknown-unknown`

```bash
error: invalid suffix `bit_target` for number literal
  --> src/lib.rs:35:13
   |
35 |     feature(32bit_target)
   |             ^^^^^^^^^^^^ invalid suffix `bit_target`
   |
   = help: the suffix must be one of the numeric types (`u32`, `isize`, `f32`, etc.)

error[E0277]: the trait bound `u64: BitStore` is not satisfied
   --> src/visitor/bit_sequence.rs:130:10
    |
130 |     U64Lsb0(BitVec<u64, Lsb0>),
    |             ^^^^^^^^^^^^^^^^^ the trait `BitStore` is not implemented for `u64`
    |
    = help: the following other types implement trait `BitStore`:
              u16
              u32
              u8
              usize
note: required by a bound in `BitVec`
   --> /home/niklasad1/.cargo/registry/src/github.com-1ecc6299db9ec823/bitvec-1.0.1/src/vec.rs:56:5
    |
56  |     T: BitStore,
    |        ^^^^^^^^ required by this bound in `BitVec`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `scale-decode` due to 2 previous errors
```

I think the ` feature(32bit_target)` is invalid or at least I have never heard of it before.

/cc @jsdw

**Comment by jsdw** [comment]
Yup, I have a plan to resolve this :)

(Basically I'd like to remove bitvec from our dependencies and provide a basic alternate decode target for them, which would also consolidate a bunch of bitvec specific stuff we have spread around the crates)

**Comment by jsdw** [comment]
Closed by #5
